### ISSUE TITLE
Compile with typescript declaration true

### ISSUE THREAD

**Issue by guilhermegregio** [emotion-js/emotion#707]
- `emotion` version: 9.2.1
- `react` version: 16.3.2

What you did:
I created a ui library with Typescript, and build not work with tsc

my tsconfig.json file
```json
{
  "compilerOptions": {
    "outDir": "build/lib",
    "baseUrl": "src",
    "rootDir": "src",
    "module": "commonjs",
    "target": "es5",
    "lib": ["es6", "es5", "dom"],
    "sourceMap": true,
    "allowJs": false,
    "jsx": "react",
    "moduleResolution": "node",
    "forceConsistentCasingInFileNames": true,
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noImplicitAny": true,
    "strictNullChecks": true,
    "suppressImplicitAnyIndexErrors": true,
    "noUnusedLocals": true,
    "declaration": true,
    "preserveConstEnums": true
  },
 ...
}

```

What happened:
I use declaration: true in tsconfig.json file to generate the type definition for my components but the compiler fails with the following message
```sh
node_modules/create-emotion-styled/types/index.d.ts:5:8 - error TS1192: Module '"../node_modules/@types/react/index"' has no default export.

5 import React, { ComponentClass, ReactHTML, ReactSVG, Ref, SFC } from 'react';
         ~~~~~


node_modules/create-emotion-styled/types/react.d.ts:4:8 - error TS1192: Module '"../node_modules/@types/react/index"' has no default export.

4 import React, { ComponentClass, Ref, SFC } from 'react';
```

**Comment by Ailrun** [comment]
Thank you for reporting issue, @guilhermegregio! Could you try `allowSyntheticDefaultImports` option and let me know whether it works or not?

**Comment by guilhermegregio** [comment]
@Ailrun tks this work to build, but when i use my lib in other project not work.

Run return this error:
```
../node_modules/ems-ui/node_modules/emotion/types/index.d.ts' not found.
```
Any suggestions?

**Comment by Ailrun** [comment]
Could you give me a minimal repoducible repo? I cannot get anything from the error...
The file does exist, doesn't it?

**Comment by guilhermegregio** [comment]
I create this repo 
https://github.com/guilhermegregio/test-ui-lib

see if you can help me.

**Comment by Ailrun** [comment]
@guilhermegregio 
In your repository, `tsconfig.json` has problem so that it cannot find valid `node_modules` directory.
When I change `packages/ui-lib/tsconfig.json` to use `"outDir": "build",` and `packages/ui-lib/package.json` with `"main": "build/index.js", "types": "build/index.d.ts",`,

`npm run start` (or `npm start`) emits different error, namely, about default import of react.
This also can be resolved with `"allowSyntheticDefaultImports": true,` in `packages/use-lib/tsconfig.json`.

However, this propagation of `"allowSyntheticDefaultImports": true,` does not look so good, so removing this would be preferable.

**Comment by guilhermegregio** [comment]
@Ailrun 
 i ran with your changes of allowSyntheticDefaultImports and it worked fine to me.

**Comment by Ailrun** [comment]
@guilhermegregio My PR is published as v9.2.3. Could you check it? If it works fine, could you close this issue?

**Comment by guilhermegregio** [comment]
@Ailrun i ran with v9.2.3 and it worked fine to me. Tks.
